external-crates/move/crates/move-compiler/src/parser/mod.rs

```diff
@@ -171,7 +171,10 @@ fn parse_file(
         Ok(()) => &source_buffer,
     };
     let (defs, comments) = match parse_file_string(compilation_env, file_hash, buffer, package) {
-        Ok(defs_and_comments) => defs_and_comments,
+        Ok(defs_and_comments) => {
+            diags.extend(Diagnostics::from(compilation_env.take_parsing_error()));
```

Comment:
I didn't notice this before and its weird as hell. Can you remove `diags` here and just put everything in the env?

Reply:
Sure thing. Through multiple changes of this code I did not notice that eventually everything ends up in the env anyways...